packages/medusa/src/utils/authenticate-middleware.ts

```diff
@@ -67,7 +67,21 @@ export const authenticate = (
       }
     }
 
-    if (authUser) {
+    const isMedusaScope =
+      stringEqualsOrRegexMatch(authScope, "admin") ||
+      stringEqualsOrRegexMatch(authScope, "store")
+
+    const isRegistered =
+      !isMedusaScope ||
+      (authUser?.app_metadata?.user_id &&
+        stringEqualsOrRegexMatch(authScope, "admin")) ||
+      (authUser?.app_metadata?.customer_id &&
+        stringEqualsOrRegexMatch(authScope, "store"))
```

Comment:
thought: Does this clash with our goal of using this middleware for more than just customers and users?

For example, if I create a new entity `Agent` that stores an `agent_id` in my `app_metadata`, and I would like agents to have admin access too. How would that be achieved?

Reply:
the agent auth_user would have to have a medusa user still. We have to validate for admin that a user exists though. If you have registered custom scope  that's not "store" or "admin" we expect the person developing the extension to manually handle validation of the auth_user's logged in status

With access control in the future this would be more easily handled but for now I think we need to ensure that a user exists. ie. an invite has been accepted